### PR TITLE
Update release docs with environment setup requirements

### PR DESCRIPTION
**Pull request by lresende**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/developer_guide/release.md [hunk 1]
```diff
@@ -18,6 +18,49 @@ limitations under the License.
...
+  pypi
+
+[pypi]
+repository:  https://upload.pypi.org/legacy/
+username: <USERNAME>
+password: <PASSWORD>
+```
+
+We use [twine](https://twine.readthedocs.io/en/latest/#installation) for uploading packages to PyPi, and another option
+to setup your credentials is to use [twine's keyring support](https://twine.readthedocs.io/en/latest/#keyring-support) for twine.
```
Comment (kiersten-stokes) [inline docs/source/developer_guide/release.md:43]:
```suggestion
to setup your credentials is to use [twine's keyring support](https://twine.readthedocs.io/en/latest/#keyring-support).
```

**Review by kiersten-stokes — APPROVED** [review]
This covers just about everything that I had in my notes! The only thing I would consider adding would be something about making sure you've accepted invitations to be a maintainer of all the correct elyra packages for PyPi and a member of the elyra org on npmjs. That's somewhat obvious though and I could go either way on it.

**Review by ptitzler — COMMENTED** [review]
```
s/PyPi/PyPI/g
```

**Review by ptitzler — COMMENTED** [review]
See earlier comment about proper camel casing of PyPI

File: docs/source/developer_guide/release.md [hunk 2]
```diff
@@ -18,6 +18,49 @@ limitations under the License.
...
+```
+
+Follow [these steps](https://docs.npmjs.com/creating-and-viewing-access-tokens) to create your authorization token.
+
+### Configuring signing keys
+
+During publishing, the packages will be signed and that will require you to have support for [pgp](https://gpgtools.org/)
+and a valid [signing key](https://gpgtools.tenderapp.com/kb/how-to/first-steps-where-do-i-start-where-do-i-begin-setup-gpgtools-create-a-new-key-your-first-encrypted-mail)
+
+Note: Although this is probably not recommended, the release scripts expects a signing key that would work without
```
Comment (kevin-bates) [inline docs/source/developer_guide/release.md:62]:
```suggestion
Note: Although this is probably not recommended, the release script expects a signing key that would work without
```

**Comment by lresende** [comment]
> See earlier comment about proper camel casing of PyPI

I don't see any other `text` without the proper case. Could you point to the culprits?

**Comment by lresende** [comment]
> > See earlier comment about proper camel casing of PyPI
> 
> I don't see any other `text` without the proper case. Could you point to the culprits?

And of course, my push was having issues and the most up-to-date changes were not in Github. 
It should be all updated now.

**Review by kiersten-stokes — APPROVED** [review]
LGTM!

**Pull request merged**